src/main/kotlin/com/team4099/robot2023/commands/drivetrain/TargetPoseCommand.kt

```diff
@@ -0,0 +1,156 @@
+package com.team4099.robot2023.commands.drivetrain
+
+import com.team4099.lib.logging.LoggedTunableValue
+import com.team4099.lib.math.asPose2d
+import com.team4099.robot2023.config.constants.DrivetrainConstants
+import com.team4099.robot2023.config.constants.FieldConstants
+import com.team4099.robot2023.subsystems.drivetrain.drive.Drivetrain
+import com.team4099.robot2023.subsystems.superstructure.Request
+import com.team4099.robot2023.util.FMSData
+import com.team4099.robot2023.util.FrameCoordinate
+import com.team4099.robot2023.util.FrameType
+import com.team4099.robot2023.util.driver.DriverProfile
+import edu.wpi.first.math.Matrix
+import edu.wpi.first.math.kinematics.Odometry
+import edu.wpi.first.wpilibj.DriverStation
+import edu.wpi.first.wpilibj.RobotBase
+import edu.wpi.first.wpilibj2.command.Command
+import org.littletonrobotics.junction.Logger
+import org.team4099.lib.controller.ProfiledPIDController
+import org.team4099.lib.controller.TrapezoidProfile
+import org.team4099.lib.geometry.Pose2d
+import org.team4099.lib.units.Velocity
+import org.team4099.lib.units.base.inMeters
+import org.team4099.lib.units.derived.Angle
+import org.team4099.lib.units.derived.Radian
+import org.team4099.lib.units.derived.degrees
+import org.team4099.lib.units.derived.inDegrees
+import org.team4099.lib.units.derived.inDegreesPerSecondPerDegree
+import org.team4099.lib.units.derived.inDegreesPerSecondPerDegreePerSecond
+import org.team4099.lib.units.derived.inDegreesPerSecondPerDegreeSeconds
+import org.team4099.lib.units.derived.inRadians
+import org.team4099.lib.units.derived.perDegree
+import org.team4099.lib.units.derived.perDegreePerSecond
+import org.team4099.lib.units.derived.perDegreeSeconds
+import org.team4099.lib.units.derived.radians
+import org.team4099.lib.units.inDegreesPerSecond
+import org.team4099.lib.units.perSecond
+import kotlin.math.PI
+import kotlin.math.atan2
+
+class TargetPoseCommand(
+    val driver: DriverProfile,
+    val driveX: () -> Double,
+    val driveY: () -> Double,
+    val turn: () -> Double,
+    val slowMode: () -> Boolean,
+    val drivetrain: Drivetrain,
+    targetCoordinate: FrameCoordinate
+) : Command() {
+    private val thetaPID: ProfiledPIDController<Radian, Velocity<Radian>>
+    private val fieldVelocitySupplier = { drivetrain.fieldVelocity.x to drivetrain.fieldVelocity.y }
+    val targetCoordinate = if (FMSData.allianceColor == DriverStation.Alliance.Red && targetCoordinate is FrameCoordinate.FieldCoordinate) {
+        FrameCoordinate.FieldCoordinate(FieldConstants.fieldLength - targetCoordinate.x, targetCoordinate.y)
```

Comment:
this will not work because it's being run on command construction (init) not when it's scheduled (initialize). do the alliance flip stuff in `initialize()`